sdks/universal-router-sdk/src/utils/constants.ts

```diff
@@ -1,129 +1,285 @@
-    router: '0x4Dae2f939ACf50408e13d58534Ff8c2776d45265',
     weth: '0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7',
-    creationBlock: 40237257,
+    routerConfigs: {
+      [RouterVersion.V1_2]: {
+        address: '0x4Dae2f939ACf50408e13d58534Ff8c2776d45265',
+        creationBlock: 40237257,
+      },
+      [RouterVersion.V2_0]: {
+        address: '0x4Dae2f939ACf50408e13d58534Ff8c2776d45265',
+        creationBlock: 40237257,
+      },
+    },
   },
   // base goerli
   [84531]: {
-    router: '0xd0872d928672ae2ff74bdb2f5130ac12229cafaf',
     weth: '0x4200000000000000000000000000000000000006',
-    creationBlock: 6915289,
+    routerConfigs: {
+      [RouterVersion.V1_2]: {
+        address: '0xd0872d928672ae2ff74bdb2f5130ac12229cafaf',
+        creationBlock: 6915289,
+      },
+      [RouterVersion.V2_0]: {
+        address: '0xd0872d928672ae2ff74bdb2f5130ac12229cafaf',
+        creationBlock: 6915289,
+      },
+    },
   },
   // base mainnet
   [8453]: {
-    router: '0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD',
     weth: '0x4200000000000000000000000000000000000006',
-    creationBlock: 9107268,
+    routerConfigs: {
+      [RouterVersion.V1_2]: {
+        address: '0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD',
+        creationBlock: 9107268,
+      },
+      [RouterVersion.V2_0]: {
+        address: '0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD',
+        creationBlock: 9107268,
+      },
+    },
   },
   [81457]: {
-    router: '0x643770E279d5D0733F21d6DC03A8efbABf3255B4',
     weth: '0x4300000000000000000000000000000000000004',
-    creationBlock: 1116444,
+    routerConfigs: {
+      [RouterVersion.V1_2]: {
+        address: '0x643770E279d5D0733F21d6DC03A8efbABf3255B4',
+        creationBlock: 1116444,
+      },
+      [RouterVersion.V2_0]: {
+        address: '0x643770E279d5D0733F21d6DC03A8efbABf3255B4',
+        creationBlock: 1116444,
+      },
+    },
   },
   [7777777]: {
-    router: '0x2986d9721A49838ab4297b695858aF7F17f38014',
     weth: '0x4200000000000000000000000000000000000006',
-    creationBlock: 11832155,
+    routerConfigs: {
+      [RouterVersion.V1_2]: {
+        address: '0x2986d9721A49838ab4297b695858aF7F17f38014',
+        creationBlock: 11832155,
+      },
+      [RouterVersion.V2_0]: {
+        address: '0x2986d9721A49838ab4297b695858aF7F17f38014',
+        creationBlock: 11832155,
+      },
+    },
   },
   [324]: {
-    router: '0x28731BCC616B5f51dD52CF2e4dF0E78dD1136C06',
     weth: '0x5aea5775959fbc2557cc8789bc1bf90a239d9a91',
-    creationBlock: 12640979,
+    routerConfigs: {
+      [RouterVersion.V1_2]: {
+        address: '0x28731BCC616B5f51dD52CF2e4dF0E78dD1136C06',
+        creationBlock: 12640979,
+      },
+      [RouterVersion.V2_0]: {
+        address: '0x28731BCC616B5f51dD52CF2e4dF0E78dD1136C06',
+        creationBlock: 12640979,
+      },
+    },
   },
 }
 
-export const UNIVERSAL_ROUTER_ADDRESS = (chainId: number): string => {
+// Todo: Change `CHAIN_CONFIGS` to pull the UR address with v4
```

Comment:
should the Todo on line 20 suffice? These seem noisy. I don't think this code will change here

Reply:
addressed!